docs/LibraryEvolution.rst

```diff
@@ -409,40 +393,29 @@ polar representation::
     }
 
 and the ``x`` and ``y`` properties have now disappeared. To avoid this, the
-bodies of inlineable functions have the following restrictions:
+bodies of inlinable functions have the following restrictions:
 
 - They may not define any local types (other than typealiases).
 
-- They must not reference any ``private`` or ``fileprivate`` entities, except
-  for other declarations marked ``@inlineable``.
+- They must not reference any ``private`` or ``fileprivate`` entities.
 
 - They must not reference any ``internal`` entities except for those that have
-  been `versioned`_ and those declared ``@inlineable``. See below for a
-  discussion of versioning internal API.
-
-- In addition, no non-public, non-versioned stored constants or variables may
-  be referenced even if they are declared ``@inlineable``. (This is because
-  their storage is still considered part of the defining library.)
+  been ``versioned <versioned entity>` and those declared ``@inlinable``. See
+  below for a discussion of versioning internal API.
 
 - They must not reference any entities from the current module introduced
-  after the function was made inlineable, except under appropriate availability
+  after the function was made inlinable, except under appropriate availability
   guards.
 
-.. _versioned: #versioning-internal-api
-
 
 Default Argument Expressions
 ----------------------------
 
-Default argument expressions are implemented as ``@inlineable`` functions and
-thus are subject to the same restrictions as inlineable functions. A default
+Default argument expressions are implemented as ``@inlinable`` functions and
+thus are subject to the same restrictions as inlinable functions. A default
```

Comment:
Only if the function is public, `@usableFromInline` or `@inlinable` -- otherwise there is no restriction.

And when it's public, the restriction is stricter -- it can only reference _public_ declarations. Referencing `@usableFromInline` is not allowed.

Reply:
Whoops, yep. Thanks.